### PR TITLE
Forbid reuse of ephemeral private key

### PR DESCRIPTION
**Pull request by emanjon**
Forbid reuse of ephemeral private key.

### CODE REVIEW

File: draft-kwiatkowski-tls-ecdhe-mlkem.md [hunk 1]
```diff
@@ -194,6 +194,8 @@ The same security considerations as those described in {{hybrid}} apply to the a
 The security analysis relies crucially on the TLS 1.3 message transcript, and one cannot assume a similar
 hybridisation is secure in other protocols.
 
+For all groups, both client and server MUST NOT reuse the ECDH or the ML-KEM part of the key share. Reusing ephemeral keys relates key material in different connections that should not be related. A vulnerability (theoretical, bad implementation like no public-key validation, or side channesl) combined with reuse can enable an attacker to recover the shared secret in another connection. Reuse also have significant privacy issues as it allows passive observers to correlate different connections.  NIST's special publication 800-56ar3 {{!SP56A}} forbids reuse by requiring that "An ephemeral private key shall be used in exactly one key-establishment transaction".
```
Comment (tomato42) [inline draft-kwiatkowski-tls-ecdhe-mlkem.md:197]: s/channesl/channels/
s/Reuse also have/Reuse also has/

Comment (ghen2) [inline draft-kwiatkowski-tls-ecdhe-mlkem.md:197]: Can "in different connections" be included in the first sentence already?

There's a valid case to reuse key shares for (different combinations of) the same algorithm *in the same connection*, like the Rustls implementation is doing for X25519 and X25519MLKEM768: https://github.com/rustls/rustls/pull/2136

While this specification appears to allow it, it's not 100% unambiguous.

Comment (emanjon) [reply draft-kwiatkowski-tls-ecdhe-mlkem.md]: fixed

Comment (emanjon) [reply draft-kwiatkowski-tls-ecdhe-mlkem.md]: Good point. fixed. Agree that this is a valid use of "reuse" without any security implications.

**Comment by kriskwiatkowski** [comment]
Let's wait for a decision from TLS chairs before merging it. Reuse is bad, but it would be a better if this requirement covers more than 3 codepoints.

**Comment by kriskwiatkowski** [comment]
+1, absolutely. This is definitely needed in general.

**Comment by bwesterb** [comment]
I agree that it's good if reuse is disallowed, but we can't change that for existing key agreements immediately. Instead, I see we could require this for new key agreement such as these in this document.

**Comment by ctz** [comment]
I feel like the text already in https://datatracker.ietf.org/doc/html/draft-ietf-tls-hybrid-design covers this in more precision -- maybe this change could re-refer to that part (6-ish paragraph in section 3.2) rather than restating it?

**Comment by emanjon** [comment]
@ctz I read 3.2 of draft-ietf-tls-hybrid-design and could not see where it forbid reuse. Could you quote.

**Review by tomato42 — APPROVED** [review]

**Comment by kriskwiatkowski** [comment]
Let's handle this as much as possible in the [draft-ietf-tls-hybrid-design](https://datatracker.ietf.org/doc/html/draft-ietf-tls-hybrid-design), so that it covers other codepoints that may potentially be introduced in the future.
This document may refer the corresponding section in draft-ietf-tls-hybrid-design once text is added.

**Comment by kriskwiatkowski** [comment]
I've submitted PR with this text to draft-ietf-tls-hybrid-design 

https://github.com/dstebila/draft-ietf-tls-hybrid-design/pull/39

**Comment by kriskwiatkowski** [comment]
@emanjon  https://github.com/dstebila/draft-ietf-tls-hybrid-design/pull/39 is merged. I hope that this solves the problem.
Feel free to re-open PR if something else needs to be added.

**Pull request closed without merging**